apps/meteor/app/api/server/v1/subscriptions.ts

```diff
@@ -103,3 +107,35 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'subscriptions.exists',
+	{
+		authRequired: true,
+		validateParams: isSubscriptionsExistsProps,
+	},
+	{
+		async get() {
+			const { username, roomId, userId, roomName } = this.queryParams;
+			const [room, user] = await Promise.all([
+				Rooms.findOneByIdOrName(roomId || roomName),
+				Users.findOneByIdOrUsername(userId || username),
+			]);
+			if (!room?._id) {
+				return API.v1.failure('error-room-not-found');
+			}
+			if (!user?._id) {
+				return API.v1.failure('error-user-not-found');
+			}
+
+			if (room.broadcast && !(await hasPermissionAsync(this.userId, 'view-broadcast-member-list', room._id))) {
```

Comment:
I'm not sure if this is really necessary 🤔 @sampaiodiego can you help on this one?

Reply:
Hey @MarcosSpessatto I used this check because this same check was implemented in channels.members for member list. I thought if this check is done to retrieve member list, then it should also be implemented while checking if a user is a member of a room or not.
https://github.com/RocketChat/Rocket.Chat/blob/ab077230dac052e753c456054dc9829d4c93a8bc/apps/meteor/app/api/server/v1/channels.ts#L1060